### PR TITLE
Wiki examples: tests added

### PR DESCRIPTION
**Pull request by lucbv**
@srajama1 @ndellingwood @jjwilke
Adding a couple of tests to make sure that the examples in the wiki are actually compiling and working correctly.
This includes two new test: wiki_crsmatrix and wiki_spmv as well as some CMake logic to get things compiled and tested.

See issue #735 for additional information.

### CODE REVIEW

**Comment by lucbv** [comment]
Sorry for opening a second PR, the previous one was using my develop branch which I keep in sync with this repo to avoid conflict, etc... this new PR actually uses a feature branch.
This shall teach me that no PR opened at 2am is worth the trouble...

**Comment by srajama1** [comment]
@lucbv : Can we add these to the tutorials directory as well ?

**Comment by lucbv** [comment]
@srajama1 sure, I did not realize we had a tutorial directory and I cannot find it in our source code, is it in a different repo?

**Comment by srajama1** [comment]
It is with Kokkos Tutorial
https://github.com/kokkos/kokkos-tutorials

**Comment by crtrott** [comment]
If you want to add them to the turoial someone needs to write the slides for that.

File: CMakeLists.txt [hunk 1]
```diff
@@ -78,6 +78,7 @@ IF (KokkosKernels_INSTALL_TESTING)
   KOKKOSKERNELS_ADD_TEST_DIRECTORIES(test_common)
   KOKKOSKERNELS_ADD_TEST_DIRECTORIES(perf_test)
   KOKKOSKERNELS_ADD_TEST_DIRECTORIES(unit_test)
+  KOKKOSKERNELS_ADD_EXAMPLE_DIRECTORIES(example)
```
Comment (jjwilke) [inline CMakeLists.txt:81]: When doing install testing we force testing on. Does it make sense to do the same for examples?
````
SET(KOKKOSKERNELS_ENABLE_EXAMPLES ON)
````

File: example/wiki/sparse/KokkosSparse_wiki_crsmatrix.cpp [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        }
+      }
+      Kokkos::deep_copy(entries, entries_h);
+      Kokkos::deep_copy(values, values_h);
+    }
+
+    graph_type myGraph(entries, row_map);
+    matrix_type myMatrix("test matrix", numRows, values, myGraph);
+  }
+
```
Comment (jjwilke) [inline example/wiki/sparse/KokkosSparse_wiki_crsmatrix.cpp:85]: The examples don't print anything at the end. Is there some performance stat or some sort of message that makes sense to print so the user knows the test is running successfully?

File: CMakeLists.txt [hunk 3]
```diff
@@ -188,7 +189,7 @@ ELSE()
   KOKKOSKERNELS_ADD_TEST_DIRECTORIES(test_common)
   KOKKOSKERNELS_ADD_TEST_DIRECTORIES(perf_test)
   KOKKOSKERNELS_ADD_TEST_DIRECTORIES(unit_test)
-  #KOKKOSKERNELS_ADD_EXAMPLE_DIRECTORIES(example)
+  KOKKOSKERNELS_ADD_EXAMPLE_DIRECTORIES(example)
```
Comment (jjwilke) [inline CMakeLists.txt:192]: I would squash the two commits.

Comment (lucbv) [reply CMakeLists.txt:192]: I figured I would get comments on the CMake side at least so I felt like squashing was going to be an inevitability before the PR merges ; )

File: example/wiki/sparse/KokkosSparse_wiki_crsmatrix.cpp (hunk 2 above)
Comment (lucbv) [reply example/wiki/sparse/KokkosSparse_wiki_crsmatrix.cpp:85]: It's a fair question, at the moment if an error happens during construction the test will fail, if it reaches completion is will return 0 which ctest interprets as success.
I can add screen output that states that the matrix constructor has been called successfully, I could do the same for the graph constructor actually.

**Comment by lucbv** [comment]
@jjwilke the comment on whitespaces reminds me that I should probably run clang formatting on the two new examples. I am guilty of not doing it since I have not looked into integrating clang-format with emacs. I am guessing that there must be a script out there that modifies the formatting policy in emacs to use the local project clang format file?

**Comment by lucbv** [comment]
### kokkos-dev: spot_check
#######################################################
PASSED TESTS
#######################################################
clang-4.0.1-Pthread_Serial-release build_time=318 run_time=126
clang-7.0.1-Cuda_OpenMP-release build_time=1078 run_time=543
cuda-9.2-Cuda_OpenMP-release build_time=1677 run_time=571
gcc-5.3.0-OpenMP-release build_time=213 run_time=62
gcc-7.3.0-Serial-release build_time=188 run_time=67
intel-16.0.3-Pthread-release build_time=289 run_time=66
intel-16.0.3-Serial-release build_time=297 run_time=75
intel-17.0.1-OpenMP-release build_time=427 run_time=54

### kokkos-dev: spot_check_tpls
#######################################################
PASSED TESTS
#######################################################
clang-4.0.1-Pthread_Serial-release build_time=315 run_time=129
clang-7.0.1-Cuda_OpenMP-release build_time=1084 run_time=263
cuda-9.2-Cuda_OpenMP-release build_time=1150 run_time=297
gcc-5.3.0-OpenMP-release build_time=216 run_time=64
gcc-7.3.0-Serial-release build_time=186 run_time=67
intel-16.0.3-Pthread-release build_time=283 run_time=65
intel-16.0.3-Serial-release build_time=300 run_time=74
intel-17.0.1-OpenMP-release build_time=424 run_time=52

**Comment by lucbv** [comment]
@ndellingwood if that's OK with you it seems that this can be merged.
Work on incorporating some of this into the kokkos-kernels tutorial will need to go into another PR against the appropriate repo.

**Comment by lucbv** [comment]
@srajama1 @brian-kelley 
If one of you feels like clicking the merge button that would be nice : )
I am working on more of these examples for another PR and it would be nice to have the CMake changes in develop.

**Pull request merged**